source/common/chromium_url/url_canon.h

```diff
@@ -145,11 +146,11 @@ template <typename T, int fixed_capacity = 1024> class RawCanonOutputT : public
   }
 
   void Resize(int sz) override {
-    T* new_buf = new T[sz];
-    memcpy(new_buf, this->buffer_, sizeof(T) * (this->cur_len_ < sz ? this->cur_len_ : sz));
+    Envoy::MemBlockBuilder<T> new_buf(sz);
+    new_buf.appendData(absl::Span<T>(this->buffer, std::min(this->cur_len_, sz)));
```

Comment:
Yeah I think this is a little better than the previous state because although it's not any fewer lines of code it encapsulates the relationship between the allocation of the temporary buffer and the safe insertion into it using a common class. Thanks.